ggml-sycl.cpp

```diff
@@ -15276,24 +15279,38 @@ static void ggml_sycl_mul_mat_batched_sycl(const ggml_tensor *src0,
 
     dpct::library_data_t cu_compute_type = dpct::library_data_t::real_float;
     dpct::library_data_t cu_data_type = dpct::library_data_t::real_float;
+    if (no_mixed_dtypes) {
+        cu_compute_type = dpct::library_data_t::real_half;
+        cu_data_type = dpct::library_data_t::real_half;
+    }
 
     // dst strides
     size_t nbd2 = dst->nb[2];
     size_t nbd3 = dst->nb[3];
 
+    const float alpha_f32 = 1.0f;
+    const float beta_f32 = 0.0f;
+
     const sycl::half alpha_f16 = 1.0f;
     const sycl::half beta_f16 = 0.0f;
 
-    const float alpha_f32 = 1.0f;
-    const float beta_f32  = 0.0f;
-
     const void * alpha = &alpha_f32;
     const void * beta  = &beta_f32;
+    if (no_mixed_dtypes) {
```

Comment:
better to have a compile time flow rather than runtime segregation?

Reply:
I chose to use the backend approach as macro pre-processors are not good standard practice for modern C++ 17. Plus the increased runtime cost is minimal as it simply an enum check. 

In general, Id like to see us move towards the case where we can do single source multiple targets and not have the compiler time target impact CPU code (I'm fine with device target macro's within the kernel due to it being a separate compilation pass in DPC++, though this is a separate discussion)